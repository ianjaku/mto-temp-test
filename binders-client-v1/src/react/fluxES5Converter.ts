/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
