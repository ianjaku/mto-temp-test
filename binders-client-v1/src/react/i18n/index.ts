import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
