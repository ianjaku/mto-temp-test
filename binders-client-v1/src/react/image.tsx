/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
