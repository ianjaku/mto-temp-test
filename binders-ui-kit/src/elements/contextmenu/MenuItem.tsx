import * as React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Icon from "../icons";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import MenuItem from "@material-ui/core/MenuItem";
import { Tooltip } from "@material-ui/core";
import autobind from "class-autobind";
import colors from "../../variables";
import cx from "classnames";
import { isIE } from "@binders/client/lib/react/helpers/browserHelper";
import "./contextmenu.styl";


export const menuItemOptionClass = "contextMenu-item-option";

export interface IContextMenuItemProps extends React.HTMLProps<HTMLInputElement> {
    classes?: {
        iconRoot: string;
        listItemIconRoot: string;
        listItemPrimary: string;
        listItemRoot: string;
        menuItemRoot: string;
    };
    className?: string;
    disabled?: boolean;
    fontSize?: string;
    icon?: unknown;
    iconName?: string;
    onClick?: (e?: React.MouseEvent) => void;
    onClose?: () => void;
    persistent?: boolean;
    title: string;
    tooltip?: string;
    noHoverAction?: boolean;
    wrapper?: (children: React.ReactNode) => React.ReactNode;
    testId?: string;
}

const swallowClick = e => e.stopPropagation();

class ContextMenuItem extends React.Component<IContextMenuItemProps, Record<string, unknown>> {
    private static defaultProps = {
        persistent: false,
    };

    constructor(props) {
        super(props);
        autobind(this, ContextMenuItem.prototype);
    }

    public render() {
        const { tooltip } = this.props;
        return !tooltip ?
            this.renderMenuItem() :
            (
                <Tooltip title={tooltip}>
                    <div onClick={swallowClick}>
                        {this.renderMenuItem()}
                    </div>
                </Tooltip>
            );
    }

    private renderMenuItem() {
        const { classes, className, disabled, style, noHoverAction, wrapper, testId } = this.props;
        const menuItem = (
            <MenuItem
                classes={{
                    root: cx(
                        "contextMenu-item",
                        { "is-disabled": disabled },
                        { "noHoverAction": noHoverAction },
                        className,
                        classes.menuItemRoot,
                    ),

                }}
                disabled={disabled}
                onClick={this.onClick}
                style={style}
                data-testid={testId}
            >
                {this.renderIcon()}
                {this.renderText()}
            </MenuItem>
        );
        return wrapper ? wrapper(menuItem) : menuItem;

    }

    private renderIconByName(name: string, classes) {
        return name && (
            <Icon
                rootClassName={`contextMenu-item-icon ${classes.iconRoot}`}
                name={name}
            />
        );
    }

    private renderIcon() {
        const { classes, icon, iconName } = this.props;
        const renderedIcon = icon || this.renderIconByName(iconName, classes);
        return renderedIcon && (
            <ListItemIcon classes={{ root: classes.listItemIconRoot }}>
                {renderedIcon}
            </ListItemIcon>
        );
    }

    private renderText() {
        const { classes, title } = this.props;

        return (
            <ListItemText
                classes={{
                    primary: `${menuItemOptionClass} ${classes.listItemPrimary}`,
                    root: `${menuItemOptionClass} ${classes.listItemRoot}`,
                }}
                primary={title}
            />
        );
    }

    private onClick(e) {
        e.stopPropagation();
        const { onClick, onClose, persistent, disabled } = this.props;
        if (disabled) {
            return;
        }
        onClick?.(e);
        if (!persistent) {
            onClose?.();
        }
    }
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const useStyles = makeStyles((theme: Theme) => createStyles({
    iconRoot: {
        fontSize: "20px",
        margin: "5px",
        top: "2px",
    },
    listItemIconRoot: {
        minWidth: "48px",
    },
    listItemPrimary: {
        fontFamily: colors.defaultFontName,
        fontSize: "13px",
        paddingRight: "50px",
        lineHeight: isIE() ? "35px" : undefined,
    },
    listItemRoot: {
        lineHeight: "35px",
        marginLeft: "0px",
        marginRight: "8px",
        minHeight: "35px",
    },
    menuItemRoot: {
        maxHeight: isIE() ? undefined : "35px",
        minHeight: isIE() ? undefined : "35px",
        paddingBottom: "0px",
        paddingTop: "0px",
    },
}));

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default React.forwardRef((props: IContextMenuItemProps, ref: any) => (
    <ContextMenuItem
        {...props}
        classes={useStyles()}
        ref={ref}
    />
));
