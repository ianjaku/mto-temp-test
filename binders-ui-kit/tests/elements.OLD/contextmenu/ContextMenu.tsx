import * as React from "react";
import ContextMenu from "../../../src/elements/contextmenu";
import MenuItem from "../../../src/elements/contextmenu/MenuItem";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

let exampleValue = 100;
const onClick = () => exampleValue++;


function createMenu(opened = false) {
    return (
        wrapWithTheme(<ContextMenu defaultOpened={opened} menuIconName={"rowing"}>
            <MenuItem onClick={onClick} title="Add document" iconName={"home"} />
            <MenuItem onClick={onClick} title="Edit" iconName={"edit"} />
            <MenuItem onClick={onClick} title="Delete" iconName={"delete"} />
            <MenuItem onClick={onClick} title="Move" iconName={"folder"} />
            <MenuItem onClick={onClick} title="Create instance" iconName={"content_copy"} />
            <MenuItem onClick={onClick} title="Share" iconName={"share"} />
            <MenuItem onClick={onClick} title="Access" iconName={"person"} />
        </ ContextMenu>)
    );

}

function createSimpleMenu() {
    return (
        wrapWithTheme(<ContextMenu defaultOpened={true}>
            <MenuItem onClick={onClick} title="Add document" iconName={"home"} />
            <MenuItem onClick={onClick} title="Edit" iconName={"edit"} />
            <MenuItem onClick={onClick} title="Delete" iconName={"delete"} />
            <MenuItem onClick={onClick} title="Move" iconName={"folder"} />
            <MenuItem onClick={onClick} title="Create instance" iconName={"content_copy"} />
            <MenuItem onClick={onClick} title="Share" iconName={"share"} />
            <MenuItem onClick={onClick} title="Access" iconName={"person"} />
        </ ContextMenu>)
    );

}

describe("Button", () => {

    beforeEach(() => {
        exampleValue = 100;
    });


    test("Base ContextMenu (snapshot)", () => {
        const element = create(createMenu());
        const serialized = element.toJSON();
        expect(serialized).toMatchSnapshot();
    });


    test("Clicking icon (mount: enzyme+jsdom)", () => {
        const element = mount(createMenu());
        element.find("button").first()
            .simulate("click");

        expect(element.find("button").hasClass("contextMenu-icon--active"));

    });

    test("Showing menu (mount: enzyme+jsdom)", () => {
        const element = mount(createSimpleMenu());
        const menu = element.find(".contextMenu");
        expect(menu.exists()).toEqual(true);

    });

    test("Closing menu (mount: enzyme+jsdom)", () => {
        const element = mount(createSimpleMenu());
        const menuItem = element.find(".contextMenu-item-icon.material-icons").at(0);
        menuItem.simulate("mouseDown");
        expect(element.find(".contextMenu-container").exists()).toEqual(false);

    });

    test("Clicking menuItem (mount: enzyme+jsdom)", () => {
        const element = mount(createSimpleMenu());
        const menuItem = element.find(".contextMenu-item-icon.material-icons").at(0);
        menuItem.simulate("mouseDown");
        expect(exampleValue).toEqual(101);

    });
});
