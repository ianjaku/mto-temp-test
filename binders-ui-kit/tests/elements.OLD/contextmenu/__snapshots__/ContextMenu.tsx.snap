// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Base ContextMenu (snapshot) 1`] = `
<div
  className=""
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "display": "inline-block",
      "muiPrepared": true,
      "position": "relative",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="contextMenu-icon material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontSize": 24,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        rowing
      </span>
    </div>
  </button>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
